catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -154,6 +151,31 @@ public MimeTypeMapper getMimeTypeMapper() {
     public void setMimeTypeMapper(MimeTypeMapper mimeTypeMapper) {
         this.mimeTypeMapper = mimeTypeMapper;
     }
+    
+    public void setRootResourceDirectories(Set<String> rootResourceDirectoryPaths) {
+        if (rootResourceDirectoryPaths == null || rootResourceDirectoryPaths.isEmpty()) {
+            LOGGER.debug("rootResourceDirectoryPaths is null or empty. Clearing rootResourceDirectories.");
+            this.rootResourceDirectories.clear();
```

Comment:
If rootResourceDirectories is null, then this is a null pointer exception at this point and if it is empty then clearing it does nothing.